File: 78. Subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> res;
        vector<int> subset;
        res.push_back(subset);
        subsets(res, subset, 0, 1, nums);
        return res;        
    }
private:
    void subsets(vector<vector<int>>& res, vector<int> subset, int begin, int length, vector<int>& nums){
        if (length == nums.size() + 1)
            return;
        for (int i = begin; i < nums.size(); i++){
            subset.push_back(nums[i]);
            res.push_back(subset);
            subsets(res, subset, i + 1, length + 1, nums);
            subset.pop_back();
        }
    }
};

File: Python/78. Subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        def subset(nums, current, ans):            
            ans.append(current.copy())
            for i in range(len(nums)):
                current.append(nums[i])
                subset(nums[i + 1:], current, ans)
                del current[-1]
        ans = list()
        subset(nums, [], ans)
        return ans
            
            
            